cmd/mothership/local/options.go

```diff
@@ -112,33 +113,66 @@ func componentsFromStrings(list []string, values []string) ([]*keb.Component, er
 	return comps, nil
 }
 
+func componentsFromClusterState(cluster cluster.State, values []string) ([]*keb.Component, error) {
+	vals := map[string]interface{}{}
+	for _, value := range values {
+		err := strvals.ParseIntoString(value, vals)
+		if err != nil {
+			return nil, fmt.Errorf("can't parse value %s", value)
+		}
+	}
+
+	components := cluster.Configuration.Components
+	for i := range components {
+		name := components[i].Component
+		if vals[name] != nil {
+			mapValue, ok := vals[name].(map[string]interface{})
+			if ok {
+				for key, value := range mapValue {
```

Comment:
Second inner loop with double `if` statement may be hard to read - what about decomposing it into smaller functions?

Reply:
This part of the code is almost the same as the [origin](https://github.com/kyma-incubator/reconciler/pull/632/files#diff-7aa83bd90c022a0af21cae41a9a134bff0edb05889da94b4d544b2e558c0cd54L93-L102) on which I based my code. I think that consistency is more important in this context. Later we can think about improving both parts